File: frontend/src/types/auditLog.ts
export interface AuditLogEntry {
  id: string
  actor_id: string | null
  actor_email: string
  target_user_id: string | null
  target_email: string | null
  action: string
  details: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLogEntry[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

File: backend/app/schemas/audit_log.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    actor_id: uuid.UUID | None
    actor_email: str
    target_user_id: uuid.UUID | None
    target_email: str | None
    action: str
    details: str | None
    created_at: datetime


class AuditLogListResponse(BaseModel):
    items: list[AuditLogRead]
    total: int
    page: int
    page_size: int
    total_pages: int